### PR TITLE
tests: Remove compatibility code not needed now when we're on Python 3

### PR DESCRIPTION
**Pull request by practicalswift**
Remove compatibility code not needed now when we're on Python 3.

### CODE REVIEW

**Comment by Empact** [comment]
utACK b95f9a6

**Comment by laanwj** [comment]
Nice cleanup.
utACK b95f9a61e0b2ae6cab0ea5431d7ee1841b80a8c8

**Comment by practicalswift** [comment]
@cdecker `rpcauth.py` has a Python 3 hashbang: `#!/usr/bin/env python3` :-)

**Comment by laanwj** [comment]
I'm ok with consistently droping Python 2 support. Yes, `rpcauth.py` just happens to work right now, but no one is testing or maintaining that\* so it'll be broken anyway at some random point in the future otherwise.

\* even worse - we don't have tests for that script at all!

**Comment by MarcoFalke** [comment]
We don't test it with python3 either. ;)

utACK b95f9a6. Affected persons can run `git revert b95f9a6` or similar.

**Comment by laanwj** [comment]
> We don't test it with python3 either. ;)

Created an issue to track that;  #12995

File: test/functional/test_framework/script.py [hunk 1]
```diff
@@ -405,17 +396,17 @@ class CScript(bytes):
...
         elif isinstance(other, CScriptNum):
             if (other.value == 0):
-                other = bchr(CScriptOp(OP_0))
+                other = bytes([CScriptOp(OP_0)])
             else:
                 other = CScriptNum.encode(other)
         elif isinstance(other, int):
             if 0 <= other <= 16:
-                other = bytes(bchr(CScriptOp.encode_op_n(other)))
+                other = bytes(bytes([CScriptOp.encode_op_n(other)]))
```
Comment (jnewbery) [inline test/functional/test_framework/script.py]: `bytes()` is idempotent. Can change `bytes(bytes())` to be `bytes()`. Same below.

**Review by jnewbery — COMMENTED** [review]
utACK b95f9a61e0b2ae6cab0ea5431d7ee1841b80a8c8. One nit.

**Pull request merged**